media-libs/amf-headers/amf-headers-1.4.34.ebuild

```diff
@@ -0,0 +1,33 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/GPUOpen-LibrariesAndSDKs/AMF"
+else
+	SRC_URI="https://github.com/GPUOpen-LibrariesAndSDKs/AMF/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="amd64 ~arm64"
+fi
+
+DESCRIPTION="The Advanced Media Framework (AMF) SDK"
+HOMEPAGE="https://github.com/GPUOpen-LibrariesAndSDKs/AMF"
+
+S="${WORKDIR}/AMF-${PV}"
+
+LICENSE="MIT"
+SLOT="0/34"
```

Comment:
You're adding a subslot but removing the subslot binder from **ffmpeg**. Something doesn't add up here.

Reply:
Thanks for highlighting this, I think we can reach the proper solution once things are properly explained.

First, you [did this](https://github.com/gentoo/gentoo/pull/33234#discussion_r1367842791) because you noticed there are ABI changes within `amdgpu-pro-amf` 
- Slotted `amdgpu-pro-amf`
- Added slot operator in ffmpeg on `amdgpu-pro-amf`

Some things you are not aware of,
- `ffmpeg` only `PDEPEND`s on `amdgpu-pro-amf`, it actually builds against `amf-headers` instead. So if it needs to rebuild on something, it should be those
- The implementation (shared libs) of the headers given by  `amf-headers` is in `amdgpu-pro-amf`
- `amf-headers` are said to be backward compatible, so I assume that even if there is version mismatches between `ffmpeg`'s built-against `amf-headers` version and `amdgpu-pro-amf` version, it should still work. 
  - As I understand it, we do not have to _strictly_ rebuild `ffmpeg` on either `amdgpu-pro-amf` nor `amf-headers` updates.
  - But probably we should just have a subslot for both follow the same one for both.